cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -122,30 +122,23 @@ runQueryProtocolParameters
   -> NetworkId
   -> Maybe OutputFile
   -> ExceptT ShelleyQueryCmdError IO ()
-runQueryProtocolParameters anyEra@(AnyCardanoEra era) anyCmodeParams@(AnyConsensusModeParams cModeParams) network mOutFile
-  | ShelleyBasedEra era' <- cardanoEraStyle era = do
-      SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr
-                               readEnvSocketPath
-      eraInMode <- hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch anyEra anyCmodeParams) $ toEraInMode cModeParams era
-
-      let localNodeConnInfo = NewIPC.LocalNodeConnectInfo
-                                { NewIPC.localConsensusModeParams = cModeParams
-                                , NewIPC.localNodeNetworkId = network
-                                , NewIPC.localNodeSocketPath = sockPath
-                                }
-          qInMode = NewIPC.QueryInEra eraInMode (NewIPC.QueryInShelleyBasedEra era' NewIPC.QueryProtocolParameters)
-      tip <- liftIO $ NewIPC.getLocalChainTip localNodeConnInfo
-      res <- liftIO $ NewIPC.queryNodeLocalState localNodeConnInfo tip qInMode
-      case res of
-        Left acqFailure -> left $ ShelleyQueryCmdAcquireFailure acqFailure
-        Right ePparams ->
-          case ePparams of
-            Left err -> left . ShelleyQueryCmdLocalStateQueryError $ EraMismatchError err
-            Right pparams -> writeProtocolParameters mOutFile pparams
-
-  | otherwise = throwError (ShelleyQueryCmdLocalStateQueryError
-                              ByronProtocolNotSupportedError)
-
+runQueryProtocolParameters anyEra@(AnyCardanoEra era) anyCmodeParams@(AnyConsensusModeParams cModeParams)
+                           network mOutFile = do
+  SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr
+                           readEnvSocketPath
+  eraInMode <- hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch anyEra anyCmodeParams) $ toEraInMode cModeParams era
+
+  let localNodeConnInfo = NewIPC.LocalNodeConnectInfo cModeParams network sockPath
+      qInMode = NewIPC.createQueryInMode eraInMode NewIPC.QueryProtocolParameters
```

Comment:
I think we can avoid using `createQueryInMode` here, which has the problem that it doesn't know the era is a Shelley era.

We can use `cardanoEraStyle era` right here, and fail in the Byron case, and otherwise we get a `ShelleyBasedEra` directly, and we can use `QueryInEra eraInMode (QueryInShelleyBasedEra era ...)` directly.